pkg/instancegroups/rollingupdate.go

```diff
@@ -236,3 +237,13 @@ func sortGroups(groupMap map[string]*cloudinstances.CloudInstanceGroup) []string
 	sort.Strings(groups)
 	return groups
 }
+
+// exitableError inspects an error to determine if the error is
+// fatal enough that the rolling update cannot continue.
+//
+// For example, if a cluster is unable to be validated by the deadline, then it
+// is unlikely that it will validate on the next instance roll, so an early exit as a
+// warning to the user is more appropriate.
+func exitableError(err error) bool {
+	return strings.HasPrefix(err.Error(), "error validating cluster")
```

Comment:
This should be some sort of type assertion, not a check of the error string.

Reply:
Sounds good! I changed the returned error that this is checking in https://github.com/kubernetes/kops/pull/14194/commits/b8e503136a88582118559c36ce4023a9de0ce1e9 to be a specific error type instead of `fmt.Errorf` so I could do the error type checking here. Additionally, I figured it was better to return this from `maybeValidate` rather than `validateWithClusterTimeout` since this is the highest level of the chain validation call stack. I can also see an argument for the error being more conceptually in sync with `validateWithClusterTimeout` as well, so happy to change it with whatever fits.